src/wp-includes/rest-api/endpoints/class-wp-rest-font-families-controller.php

```diff
@@ -0,0 +1,554 @@
+
+		if ( isset( $request['id'] ) ) {
+			// Allow sending individual properties if we are updating an existing font family.
+			unset( $schema['required'] );
+
+			// But don't allow updating the slug, since it is used as a unique identifier.
+			if ( isset( $settings['slug'] ) ) {
+				return new WP_Error(
+					'rest_invalid_param',
+					/* translators: %s: Name of parameter being updated: font_family_settings[slug]". */
+					sprintf( __( '%s cannot be updated.' ), 'font_family_settings[slug]' ),
+					array( 'status' => 400 )
+				);
+			}
+		}
+
+		// Check that the font face settings match the theme.json schema.
+		$has_valid_settings = rest_validate_value_from_schema( $settings, $schema, 'font_family_settings' );
+
+		if ( is_wp_error( $has_valid_settings ) ) {
+			$has_valid_settings->add_data( array( 'status' => 400 ) );
+			return $has_valid_settings;
+		}
+
+		// Check that none of the required settings are empty values.
+		foreach ( $required as $key ) {
+			if ( isset( $settings[ $key ] ) && ! $settings[ $key ] ) {
+				return new WP_Error(
+					'rest_invalid_param',
+					/* translators: %s: Name of the empty font family setting parameter, e.g. "font_family_settings[slug]". */
+					sprintf( __( '%s cannot be empty.' ), "font_family_settings[ $key ]" ),
+					array( 'status' => 400 )
+				);
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sanitizes the font family settings when creating or updating a font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string          $value   Encoded JSON string of font family settings.
+	 * @param WP_REST_Request $request Request object.
+	 * @return array                   Decoded array font family settings.
+	 */
+	public function sanitize_font_family_settings( $value ) {
+		// Settings arrive as stringified JSON, since this is a multipart/form-data request.
+		$settings = json_decode( $value, true );
+		$schema   = $this->get_item_schema()['properties']['font_family_settings']['properties'];
+
+		// Sanitize settings based on callbacks in the schema.
+		foreach ( $settings as $key => $value ) {
+			$sanitize_callback = $schema[ $key ]['arg_options']['sanitize_callback'];
+			$settings[ $key ]  = call_user_func( $sanitize_callback, $value );
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Creates a single font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_item( $request ) {
+		$settings = $request->get_param( 'font_family_settings' );
+
+		// Check that the font family slug is unique.
+		$query = new WP_Query(
+			array(
+				'post_type'              => $this->post_type,
+				'posts_per_page'         => 1,
+				'name'                   => $settings['slug'],
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			)
+		);
+		if ( ! empty( $query->get_posts() ) ) {
```

Comment:
Same here, the `WP_Query` constructor already calls `get_posts()`. This just causes duplicate work.

```suggestion
		if ( ! empty( $query->posts ) ) {
```

Reply:
I'm addressing this in this Gutenberg PR: https://github.com/WordPress/gutenberg/pull/58736. Once merged, I'll commit to this PR.